### PR TITLE
Removes Bay's Wound System

### PR DESCRIPTION
**Pull request by Fox-McCloud**
FUCK YOUR CUNT YOU SHIT EATING COCKSTORM AND EAT A DONG FUCKING ASS ENDING SHITFUCK. EAT PENISES IN YOUR FUCKFACE AND SHIT OUT ABORTIONS OF FUCK AND POO AND SHIT IN YOUR ASS YOU COCK FUCK SHIT MONKEY FUCK ASS WANKER FROM THE DEPTHS OF SHIT

Removes the Bay datum wound system; moving everything to be handled on the limb itself (ie: `internal_bleeding` var on the limb, instead of creating a wound datum).

We have no reason to have this horrifyingly over-complicated system, and it allows us to kill off stupid limb stumps, while we're at it; good freaking riddance.

This also cleans up examining humans. There was a lot of confusing code here, redundancies, unnecessary info (such as the dwarf mutation being shown...though that as before it altered your actual height), and checks that were obviously tailored to the assumption all species would have blood and a pulse.

It's a lot cleaner now, better displayed, and the proper color (some things were showing up blue when they should have been purple or red).

img
img

:cl: Fox McCloud
fix: Fixes dismembering and some damage transfers (when limb damage is maxed out) being inconsistent
tweak: can no longer dismember groins
tweak: Cleans up human examine code to be better laid out and more consistent; minor grammatical changes and some fixes; consistency should be retained for species that lack a pulse. Soul departure now properly checks for if the person can re-enter their corpse or not. Also resolves issues with FAKE DEATH examination.
/:cl:

### CODE REVIEW

**Comment by SamHPurp** [comment]
image

**Comment by scrubmcnoob** [comment]
let the hate flow
https://i.imgflip.com/w120h.jpg

File: code/modules/mob/mob.dm [hunk 1]
```diff
@@ -867,19 +867,6 @@ var/list/slot_equipment_priority = list( \
 /mob/proc/stripPanelEquip(obj/item/what, mob/who)
 	return
 
-
-/mob/proc/pull_damage()
```
Comment (Fox-McCloud) [inline code/modules/mob/mob.dm]: Not used anymore; the new blood system accomplishes this, pretty much, anyway.

**Comment by taukausanake** [comment]
I see this as turning us away more from Bay and into /tg/. However more things working on the actual organ sounds nicer

File: code/modules/surgery/organs/blood.dm [hunk 2]
```diff
@@ -79,8 +80,14 @@
 			if(BP.open)
 				temp_bleed += 0.5
 
+			if(BP.internal_bleeding)
+				internal_bleeding_rate += 0.5
+
 		bleed_rate = max(bleed_rate - 0.5, temp_bleed)//if no wounds, other bleed effects (heparin) naturally decreases
 
+		if(internal_bleeding_rate && !(status_flags & FAKEDEATH))
+			bleed(internal_bleeding_rate)
```
Comment (Fox-McCloud) [inline code/modules/surgery/organs/blood.dm]: This is separate so that you can't bandage yourself and bypass the bleeding.

**Comment by Fox-McCloud** [comment]
@taukausanake 

I'm ***not*** killing off infections, bone fracturing, or internal bleeding; those will all still be around and function as closely as possible to how they are on live (there will be some differences); this is just cleaning up and out a system that is barely used, hugely complicated, and makes improving/changing the limb system way more difficult.

**Comment by tigercat2000** [comment]
@taukausanake The wound system adds very little to the actual complexity of the health system, but it makes code an absolute nightmare to work with, because it's so terribly coded that you really can't make out how it works whatsoever. It was one of the biggest bugfests when Mark originally ported Bay's organs, and it isn't that much better now.

File: code/controllers/configuration.dm [hunk 3]
```diff
@@ -96,9 +96,6 @@
 	var/health_threshold_crit = 0
 	var/health_threshold_dead = -100
 
-	var/organ_health_multiplier = 1
-	var/organ_regeneration_multiplier = 1
```
Comment (Fox-McCloud) [inline code/controllers/configuration.dm]: We've had external wound regeneration disabled for 3 years now; on top of which, with how we changed `take_overall_damage` and `heal_overall_damage` to actually do as the name implies....well....you'd be able to heal from even 70+ damage wounds.

File: code/modules/surgery/organs/wound.dm [hunk 4]
```diff
@@ -11,41 +11,20 @@
 
 	// amount of damage this wound causes
 	var/damage = 0
-	// ticks of bleeding left.
```
Comment (Crazylemon64) [inline code/modules/surgery/organs/wound.dm]: wait, am I missing something, why isn't this entire file axed

Comment (Fox-McCloud) [inline code/modules/surgery/organs/wound.dm]: This is still WIP; I need to migrate damage to be handled entirely by the limbs themselves and get rid of limb stumps.

Once that is done, and tested, this *won't* be here.

**Comment by RemieRichards** [comment]
@Fox-McCloud you have done the grand sin: getting beepsky's quote wrong.
1000 years, dungeon.

**Comment by KingPhilipIII** [comment]
@RemieRichards 

No trial.

**Comment by Vivalas** [comment]
Was the meme branch name really worth all the copy pasta gore required to reference the branch in the command line?

File: code/modules/surgery/organs/organ.dm [hunk 5]
```diff
@@ -101,11 +100,6 @@ var/list/organ_cache = list()
 		return
 
 	if(!owner)
-		if(reagents && prob(40))
-			reagents.remove_any(0.1)
-			for(var/datum/reagent/R in reagents.reagent_list)
-				R.reaction_turf(get_turf(src), 0.1)
```
Comment (Fox-McCloud) [inline code/modules/surgery/organs/organ.dm]: This is never capable of doing anything, so this is just getting removed.

File: code/modules/surgery/organs/organ_stump.dm [hunk 6]
```diff
@@ -1,28 +0,0 @@
-/obj/item/organ/external/stump
```
Comment (Fox-McCloud) [inline code/modules/surgery/organs/organ_stump.dm]: FUCK. YOU.

**Comment by Fox-McCloud** [comment]
Well, I finally figured out why people with robotic limbs are so strong.

Basically, damage that would transfer from a robotic limb to another would run through the damage resistances twice (meaning if you capped out on one limb's damage, it would get reduced damage, then, if that damage transferred to another limb, it would would get reduced...again).

On top of this, if a robotic limb would fail, it would halt further damage transfer...meaning if an esword hit a limb that was 2 damage away from capping out, and it was a robotic limb, it would deal 2 damage, then the limb would dismember and not transfer the remaining 28 damage to the parent.

This also ensures that damage of a disemembered organ *and* its children transfer to the parent organ.

No more cutting off your arm to heal damage, anymore.

**Comment by Fox-McCloud** [comment]
Ready for review.

Tested this pretty thoroughly; everything appears to work fine; bleeding, damage taking, damage healing, rejuving, dismemberment, infections, internal bleeding, broken bones, bone jostling...getting your head cut off then rejuv'd.

File: code/modules/surgery/organs/organ.dm [hunk 7]
```diff
@@ -116,9 +110,6 @@ var/list/organ_cache = list()
 		if(germ_level >= INFECTION_LEVEL_THREE)
 			necrotize()
 
-		if(damage >= max_damage)
```
Comment (Fox-McCloud) [inline code/modules/surgery/organs/organ.dm]: this is done here, then again below; this is redundant to have it necrotize twice.

**Comment by SamHPurp** [comment]
That 'fix' has balance implications because of how long its been in effect.... but I'm all for it.

**Comment by Fox-McCloud** [comment]
@RemieRichards 

Corrected Beepsky's phrasing 😛

**Comment by Fox-McCloud** [comment]
Because the wound system had to touch human examine, I cleaned that up, as well. What a nightmare.

File: code/modules/mob/living/carbon/human/examine.dm [hunk 8]
```diff
@@ -212,80 +202,32 @@
...
-		msg += "[t_He] [t_has] a stupid expression on [t_his] face.\n"
-
-	if(species.show_ssd && (!species.has_organ["brain"] || get_int_organ(/obj/item/organ/internal/brain)) && stat != DEAD)
-		if(istype(src, /mob/living/carbon/human/interactive))
-			msg += "<span class='deadsay'>[t_He] appears to be some sort of sick automaton: [t_his] eyes are glazed over and [t_his] mouth is slightly agape.</span>\n"
-		else if(!key)
-			msg += "<span class='deadsay'>[t_He] [t_is] fast asleep. It doesn't look like they are waking up anytime soon.</span>\n"
-		else if(!client)
-			msg += "[t_He] [t_has] suddenly fallen asleep, suffering from Space Sleep Disorder.\n"
+	var/appears_dead = 0
```
Comment (SamHPurp) [inline code/modules/mob/living/carbon/human/examine.dm]: Boolean? `TRUE` / `FALSE`

File: code/modules/mob/living/carbon/human/examine.dm [hunk 9]
```diff
@@ -212,80 +202,32 @@
...
+	if(stat == DEAD || (status_flags & FAKEDEATH))
+		appears_dead = 1
```
Comment (SamHPurp) [inline code/modules/mob/living/carbon/human/examine.dm]: `TRUE`

File: code/modules/mob/living/carbon/human/examine.dm [hunk 10]
```diff
@@ -212,80 +202,32 @@
...
+		if(suiciding)
+			msg += "<span class='warning'>[t_He] appears to have committed suicide... there is no hope of recovery.</span>\n"
+		msg += "<span class='deadsay'>[t_He] [t_is] limp and unresponsive; there are no signs of life"
+		if(get_int_organ(/obj/item/organ/internal/brain))
+			if(!key)
+				var/foundghost = 0
```
Comment (SamHPurp) [inline code/modules/mob/living/carbon/human/examine.dm]: Boolean? `TRUE` / `FALSE`

File: code/modules/mob/living/carbon/human/examine.dm [hunk 11]
```diff
@@ -212,80 +202,32 @@
...
+				if(mind)
+					for(var/mob/dead/observer/G in player_list)
+						if(G.mind == mind)
+							foundghost = 1
```
Comment (SamHPurp) [inline code/modules/mob/living/carbon/human/examine.dm]: `TRUE`

File: code/modules/mob/living/carbon/human/examine.dm [hunk 12]
```diff
@@ -212,80 +202,32 @@
...
+							if(G.can_reenter_corpse == 0)
```
Comment (SamHPurp) [inline code/modules/mob/living/carbon/human/examine.dm]: `if(!G.can_reenter_corpse)` no?

File: code/modules/mob/living/carbon/human/examine.dm [hunk 13]
```diff
@@ -212,80 +202,32 @@
...
+								foundghost = 0
```
Comment (SamHPurp) [inline code/modules/mob/living/carbon/human/examine.dm]: `FALSE`

File: code/modules/mob/living/carbon/human/examine.dm [hunk 14]
```diff
@@ -296,55 +238,17 @@
...
-				wound_flavor_text["[temp.limb_name]"] += " and"
-			if(temp.burn_dam)
-				switch(temp.burn_dam)
-					if(0 to 20)
-						wound_flavor_text["[temp.limb_name]"] += " some burns"
-					if(21 to INFINITY)
-						wound_flavor_text["[temp.limb_name]"] += pick(" a lot of burns"," severe melting")
-			if(wound_flavor_text["[temp.limb_name]"])
-				wound_flavor_text["[temp.limb_name]"] += "!</span>\n"
+			wound_flavor_text["[organ_tag]"] = "<b>[t_He] [t_is] missing [t_his] [organ_descriptor].</b>\n"
```
Comment (SamHPurp) [inline code/modules/mob/living/carbon/human/examine.dm]: you're inconsistent with `<b>` and `<B>` again down lower down... can we pick just one version and always use that?

File: code/modules/mob/living/carbon/human/examine.dm (hunk 12 above)
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/human/examine.dm]: Not boolean, actually.

Comment (SamHPurp) [inline code/modules/mob/living/carbon/human/examine.dm]: Bizarre. Could probably be tidied in future then.

**Pull request merged**